staging/src/k8s.io/apimachinery/pkg/util/wait/wait_test.go

```diff
@@ -439,7 +440,10 @@ func TestPollForever(t *testing.T) {
 		select {
 		case _, open := <-ch:
 			if !open {
-				t.Fatalf("did not expect channel to be closed")
+				if len(errc) != 0 {
+					t.Fatalf("did not expect channel to be closed, %v", <-errc)
```

Comment:
why not before the loop? errc is only written before ch is closed.

Reply:
Thanks for your feedback @sttts! Think it's not sufficient to check before the loop. Although `errc` is only written before `ch` is closed, that only happens after `done` has been written in [l.454](https://github.com/tiloso/kubernetes/blob/627218a8e2e8dddc70312c04b0fd5a90bccb8022/staging/src/k8s.io/apimachinery/pkg/util/wait/wait_test.go#L454). Since `PollInfinite` calls the `ConditionFunc` repeatedly, it might return an error until then.

Reply:
I missed line 418